packages/shared/trade/src/transaction-approvals/hooks/use-transaction-approvals.tsx

```diff
@@ -0,0 +1,64 @@
+import { useTokenApproval } from './use-token-approval';
+import { useEnablePrimeBorrow } from './use-enable-prime-borrow';
+import {
+  useAccountReady,
+  BaseTradeContext,
+  useWalletBalanceInputCheck,
+} from '@notional-finance/notionable-hooks';
+
+export const useTransactionApprovals = (
+  symbol: string,
+  context: BaseTradeContext,
+  isWithdraw?: boolean
+) => {
+  const {
+    state: { depositBalance, debt },
+  } = context;
+  const isAccountReady = useAccountReady();
+  const {
+    tokenStatus,
+    isSignerConnected,
+    enableToken,
+    tokenApprovalTxnStatus,
+  } = useTokenApproval(symbol);
+
+  const isDebt =
+    debt?.tokenType === 'PrimeDebt' || debt?.tokenType === 'VaultDebt'
+      ? true
+      : false;
+
+  const { isPrimeBorrowAllowed, enablePrimeBorrow, variableBorrowTxnStatus } =
+    useEnablePrimeBorrow();
+
+  const { allowance, insufficientAllowance } = useWalletBalanceInputCheck(
+    depositBalance?.token,
+    depositBalance
+  );
+
+  const variableBorrowApprovalRequired = !isPrimeBorrowAllowed && isDebt;
```

Comment:
I think the complete logic here should be:
1. Put the `enablePrimeBorrow` prop back everywhere 
`const variableBorrowApprovalRequired = !isPrimeBorrowAllowed && enablePrimeBorrow;`

Reply:
Note that enablePrimeBorrow should always be false on vault pages